netops_demo/golang/src/github.com/v3io/demos/functions/ingest/ingest.go

```diff
@@ -33,27 +33,32 @@ type userData struct {
 
 func Ingest(context *nuclio.Context, event nuclio.Event) (interface{}, error) {
 	userData := context.UserData.(*userData)
-	parsedMetrics := map[string]*metricSamples{}
+	parsedMetricsArray := [][]map[string]*metricSamples{}
 
 	// try to parse the input body
-	if err := json.Unmarshal(event.GetBody(), &parsedMetrics); err != nil {
+	if err := json.Unmarshal(event.GetBody(), &parsedMetricsArray); err != nil {
 		return nil, nuclio.NewErrBadRequest(err.Error())
 	}
 
 	// iterate over metrics
-	for metricName, metricSamples := range parsedMetrics {
-
-		// all arrays must contain same # of samples
-		if !allMetricSamplesFieldLenEqual(metricSamples) {
-			return nil, nuclio.NewErrBadRequest("Expected equal number of samples")
-		}
-
-		// iterate over values and ingest them into all time series datastores
-		if err := ingestMetricSamples(context, userData, metricName, metricSamples); err != nil {
-			return nil, nuclio.NewErrBadRequest(err.Error())
+	for _, sampleDates := range parsedMetricsArray {
```

Comment:
Regardless of my not understanding the outer list here - the argument names here seem to be a bit mixed up (sampleDates is the entry of parsedMetricsArray, parsedMetrics is the entry into sampleDates)

Reply:
Done